tests/test_containers.py

```diff
@@ -0,0 +1,173 @@
+from roborock import UserData, HomeData, Consumable, Status, DNDTimer, CleanSummary, CleanRecord
+from .mock_data import USER_DATA, HOME_DATA_RAW, CONSUMABLE, STATUS, DND_TIMER, CLEAN_SUMMARY, CLEAN_RECORD
+
+
+def test_user_data():
+    ud = UserData(USER_DATA)
+    assert ud.uid == 123456
+    assert ud.token_type == "token_type"
+    assert ud.token == "abc123"
+    assert ud.rr_uid == "abc123"
+    assert ud.region == "us"
+    assert ud.country == "US"
+    assert ud.country_code == '1'
+    assert ud.nickname == "user_nickname"
+    assert ud.rriot.user == "user123"
+    assert ud.rriot.password == "pass123"
+    assert ud.rriot.h_unknown == "unknown123"
+    assert ud.rriot.domain == "domain123"
+    assert ud.rriot.reference.region == "US"
+    assert ud.rriot.reference.api == "https://api-us.roborock.com"
+    assert ud.rriot.reference.mqtt == "ssl://mqtt-us.roborock.com:8883"
+    assert ud.rriot.reference.l_unknown == "https://wood-us.roborock.com"
+    assert ud.tuya_device_state == 2
+    assert ud.avatar_url == "https://files.roborock.com/iottest/default_avatar.png"
+
+
+def test_home_data():
+    hd = HomeData(HOME_DATA_RAW)
+    assert hd.id == 123456
+    assert hd.name == "My Home"
+    assert hd.lon is None
+    assert hd.lat is None
+    assert hd.geo_name is None
+    product = hd.products[0]
+    assert product.id == "abc123"
+    assert product.name == "Roborock S7 MaxV"
+    assert product.code == "a27"
+    assert product.model == "roborock.vacuum.a27"
+    assert product.iconurl is None
+    assert product.attribute is None
+    assert product.capability == 0
+    assert product.category == "robot.vacuum.cleaner"
+    schema = product.schema
+    assert schema[0].id == "101"
+    assert schema[0].name == "rpc_request"
+    assert schema[0].code == "rpc_request_code"
+    assert schema[0].mode == "rw"
+    assert schema[0].type == "RAW"
+    assert schema[0].product_property is None
+    assert schema[0].desc is None
+    device = hd.devices[0]
+    assert device.duid == "abc123"
+    assert device.name == "Roborock S7 MaxV"
+    assert device.attribute is None
+    assert device.activetime == 1672364449
+    assert device.local_key == "key123"
+    assert device.runtime_env is None
+    assert device.time_zone_id == "America/Los_Angeles"
+    assert device.icon_url == "no_url"
+    assert device.product_id == "product123"
+    assert device.lon is None
+    assert device.lat is None
+    assert not device.share
+    assert device.share_time is None
+    assert device.online
+    assert device.fv == "02.56.02"
+    assert device.pv == "1.0"
+    assert device.room_id == 2362003
+    assert device.tuya_uuid is None
+    assert not device.tuya_migrated
+    assert device.extra == '{"RRPhotoPrivacyVersion": "1"}'
+    assert device.sn == "abc123"
+    assert device.feature_set == "2234201184108543"
+    assert device.new_feature_set == "0000000000002041"
+    # status = device.device_status
+    # assert status.name ==
```

Comment:
device_status is not actually correct. The parameters for it are completely different than what the api returns. I don't think it used anywhere, but I noticed it from the test so I figured I would share.